src/components/dao/dao.js

```diff
@@ -287,22 +389,46 @@ export default class Dao extends Component {
                     bond: bond,
                     policy: policy
                 },
-                loading: false,
-                proposed: proposed
+                loading: false
+                // proposed & proposedInfo will be handled in the next "then()"
             }
-        )
-        .then(() => {
-            // can user propose FunctionCall to DAO?
-            const canPropose = newState.infos.policy?.roles
+            if (policy !== undefined) {
+                return this.proposalAlreadyExists(lastProposalID, policy?.proposal_period).catch(e => {});
+            }
+        })
+        .then(( result ) => {
+            if (result !== undefined) {
+                const { proposal_id, proposal_info } = result;
+                newState.proposed = proposal_id;
```

Comment:
Can u do this? Shouldn't this be `newState["proposed"] = proposal_id`, since `newState` doesn't know about that property yet?

Reply:
Apparently I can 🤷‍♂️